model/CompatibilityChecker.php

```diff
@@ -14,101 +14,222 @@
 use Sinergi\BrowserDetector\Browser;
 use Sinergi\BrowserDetector\Os;
+use tao_models_classes_FileNotFoundException;
 
-class CompatibilityChecker
+class CompatibilityChecker extends ConfigurableService
 {
+    public const SERVICE_ID = 'taoClientDiagnostic/CompatibilityChecker';
+
+    public const COMPATIBILITY_NONE = 0;
+    public const COMPATIBILITY_COMPATIBLE = 1;
+    public const COMPATIBILITY_NOT_TESTED = 2;
+    public const COMPATIBILITY_SUPPORTED = 3;
+    public const COMPATIBILITY_NOT_SUPPORTED = 4;
+
     protected $compatibility;
+    protected $supported;
+    protected $excludedBrowsers;
+    protected $excludedOS;
 
     /**
-     * CompatibilityChecker constructor
-     * Check parameter required
      * Extract compatibility file
-     * @throws \tao_models_classes_FileNotFoundException
+     * @throws tao_models_classes_FileNotFoundException
+     */
+    public function getCompatibilityList(): array
+    {
+        if (!$this->compatibility) {
+            $compatibilityFile = __DIR__ . '/../include/compatibility.json';
+
+            if (!file_exists($compatibilityFile)) {
+                throw new tao_models_classes_FileNotFoundException("Unable to find the compatibility file");
+            }
+            $this->compatibility = json_decode(file_get_contents($compatibilityFile), true);
+        }
+        return $this->compatibility;
+    }
+
+    /**
+     * Fetch the support list
+     * @throws tao_models_classes_FileNotFoundException
+     */
+    public function getSupportedList(): array
+    {
+        if (!$this->supported) {
+            $service = $this->getServiceLocator()->get(DiagnosticServiceInterface::SERVICE_ID);
+            $config = $service->getDiagnosticJsConfig();
+            $supportListUrl = $config['diagnostic']['testers']['browser']['browserslistUrl'];
+
+            if (!$supportListUrl) {
+                throw new tao_models_classes_FileNotFoundException("The URL to the list of supported browser is not configured");
+            }
+            $supportedList = json_decode(file_get_contents($supportListUrl), true);
+
+            if (!$supportedList) {
+                throw new tao_models_classes_FileNotFoundException("Unable to fetch the list of supported browsers");
+            }
+
+            $this->supported = array_map(function ($entry) {
+                $entry['compatible'] = self::COMPATIBILITY_SUPPORTED;
+
+                $entry['versions'] = array_reduce($entry['versions'], function ($versions, $version) {
+                    return array_merge($versions, explode('-', $version));
+                }, []);
+
+                return $entry;
+            }, $supportedList);
+        }
+        return $this->supported;
+    }
+
+    /**
+     * Standard version_compare threats that  5.2 < 5.2.0, 5.2 < 5.2.1, ...
+     */
+    protected function versionCompare($ver1, $ver2): int
+    {
+        $ver1 = preg_replace('#(\.0+)+($|-)#', '', $ver1);
```

Comment:
`preg_replace('#(\.0+)+($|-)#', '', $ver);` could move to another private method

Reply:
good catch 👍